### PR TITLE
Support deleting original files when converting from CLI

### PR DESCRIPTION
**Pull request by christophehenry**
This add options `-D` and `--delete-original` to support deleting original files when conversion is done. This PR does not includes translations in english and french because I don't know how to deal with `.po` files. Don't hesitate to tell me if you want this to be embedded in this PR.

Also, there's no tests as this was quite trivial. Tell me if you need them.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,5 +64,6 @@ def run(self):
     ],
     cmdclass={
         'install': Install
-    }
+    },
+    scripts=["bin/soundconverter"]
```
Comment (christophehenry) [inline setup.py:68]: For whatever reason, `bin/soundconverter` didn't get picked by `python setup.py install` on my Fedora until I added this.

File: soundconverter/interface/batch.py [hunk 2]
```diff
@@ -352,6 +352,7 @@ def __init__(self, input_files):
             converter = Converter(sound_file, name_generator)
 
             converter.existing_behaviour = settings.get('existing')
+            converter.delete_original = settings.get('delete-original')
```
Comment (sezanzeb) [inline soundconverter/interface/batch.py:355]: Check [`use_memory_gsettings`, line 67](https://github.com/kassoulet/soundconverter/blob/main/soundconverter/interface/batch.py#L67). This should probably be stored there instead to be consistent with the other settings

**Review by sezanzeb — CHANGES_REQUESTED** [review]

File: soundconverter/interface/batch.py (hunk 2 above)
Comment (sezanzeb) [reply soundconverter/interface/batch.py:355]: `existing_behaviour` seems to be only available via the batch mode. I don't remember why it is not a gio_setting as well, and it certainly causes some confusion.

Anyhow, I guess the current way of handling things if this doesn't change in that regard is: if the setting is relevant for both the gui and the batch mode, use `use_memory_gsettings`

Comment (christophehenry) [reply soundconverter/interface/batch.py:355]: I did that in the first draft but isn't it persisted across restarts?

File: soundconverter/interface/batch.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def use_memory_gsettings(options):
     gio_settings = Gio.Settings.new_with_backend('org.soundconverter', backend)
     set_gio_settings(gio_settings)
 
-    gio_settings.set_boolean('delete-original', False)
+    gio_settings.set_boolean('delete-original', options['delete-original'])
```
Comment (sezanzeb) [inline soundconverter/interface/batch.py:67]: Thanks for the change, but now tests are not passing. Try

```suggestion
    gio_settings.set_boolean('delete-original', options.get('delete-original', False))
```

Also, feel free to add a new test. You could look at `test_set_delete_original_false` in `integration.py` and make a copy of it that tests the `-D` argument. And then use something like `self.assertFalse(os.path.isfile(...))` to check if it correctly deleted the source file.

Comment (christophehenry) [reply soundconverter/interface/batch.py]: Done.

**Review by sezanzeb — APPROVED** [review]

**Comment by kassoulet** [comment]
Thank you!

**Pull request merged**